pkg/api/types.go

```diff
@@ -597,10 +597,34 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type HostPathType string
+
+const (
+	// For backwards compatible, leave it empty if unset
+	HostPathUnset HostPathType = ""
+	// If nothing exists at the given path, an empty directory will be created there.
+	HostPathDirectoryOrCreate HostPathType = "DirectoryOrCreate"
+	// A directory must exist at the given path
+	HostPathDirectory HostPathType = "Directory"
+	// If nothing exists at the given path, an empty file will be created there.
+	HostPathFileOrCreate HostPathType = "FileOrCreate"
```

Comment:
@tallclair The permission of the created directory is [0755](https://github.com/kubernetes/kubernetes/pull/46597/commits/9f0ae008115b448fbecdb76419050f5703983804#diff-def787bccf1d007b36389403ca495928R377), the created file is [0644](https://github.com/kubernetes/kubernetes/pull/46597/commits/9f0ae008115b448fbecdb76419050f5703983804#diff-def787bccf1d007b36389403ca495928R359). The owner/group is just the current owner/group, stay unchanged.